java/code/src/com/redhat/rhn/manager/system/ServerGroupManager.java

```diff
@@ -327,7 +340,8 @@ public void addServers(ServerGroup sg, Collection<Server> servers, User loggedIn
         validateAccessCredentials(loggedInUser, sg, sg.getName());
         validateAdminCredentials(loggedInUser);
 
-        SystemManager.addServersToServerGroup(servers, sg);
+        SystemManager systemManager = new SystemManager(ServerFactory.SINGLETON, ServerGroupFactory.SINGLETON, saltApi);
```

Comment:
this system manager should not be passed as a class field instead of being created on each method invocation?